File: backend/internal/core/domain/metadata.go
package domain

type Metadata struct {
	Sort         string `query:"sort" json:"sort" validate:"oneof=id user_id name title created_at updated_at"`
	Order        string `query:"order" json:"order" validate:"oneof=asc desc"`
	TotalRecords int64  `json:"total_records"`
	TotalPage    int    `json:"total_page"`
	Limit        int    `query:"limit" json:"limit"`
	Page         int    `query:"page" json:"page"`
}

File: frontend/src/lib/schema/metadata.ts
import { z } from "zod";

export const metadataSchema = z.object({
  sort: z.enum(["id", "user_id", "name", "title", "created_at", "updated_at"]),
  order: z.enum(["asc", "desc"]),
  total_records: z.number(),
  total_pages: z.number(),
  limit: z.number(),
  page: z.number(),
});

export type Metadata = z.infer<typeof metadataSchema>;
